### PR TITLE
[#1830] improvement(core,hive):  TableOperations support impersonation

### PR DESCRIPTION
**Pull request by qqqttt123**
### What changes were proposed in this pull request?
Now, we only support catalog operations to impersonate. We should support it in the TableOperation.

### Why are the changes needed?
Better user experience.

Fix: #1830 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Add UT.

### CODE REVIEW

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java [hunk 1]
```diff
@@ -124,7 +124,7 @@ public Table loadTable(NameIdentifier ident) throws NoSuchTableException {
             ErrorHandlers.tableErrorHandler());
     resp.validate();
 
-    return resp.getTable();
+    return RelationalTable.from(ident.namespace(), resp.getTable(), restClient);
```
Comment (qqqttt123) [inline clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java:127]: cc @mchades

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java [hunk 2]
```diff
@@ -207,7 +207,7 @@ public Table alterTable(NameIdentifier ident, TableChange... changes)
...
```
Comment (qqqttt123) [inline clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java:210]: cc @mchades

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java (hunk 1 above)
Comment (mchades) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java:127]: LGTM. It is indeed a legacy bug

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java (hunk 2 above)
Comment (mchades) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/RelationalCatalog.java:210]: LGTM. It is indeed a legacy bug

**Review by jerryshao — APPROVED** [review]

**Pull request merged**